sound/soc/sof/sof-audio.c

```diff
@@ -107,6 +107,17 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (!swidget->private)
 		return 0;
 
+	/*
+	 * PCM streams can be freed without being restarted after system resume. Prevent
+	 * the use_count from being decremented below 0 for dynamic widgets and below 1
+	 * for static widgets in such scenarios.
+	 */
+	if (swidget->dynamic_pipeline_widget && !swidget->use_count)
+			return 0;
```

Comment:
can we just use if (!swidget->use_count). That would prevent the counter from going into negative values in corner cases.